src/backend/services/mining-api/src/routes.py

```diff
@@ -41,72 +43,94 @@ def send_to_connected_clients(transaction):
     ongoing_proof = transaction["id"]
     ongoing_transaction = transaction
 
-    while (connected_clients == 0) : 
+    while (connected_clients == 0):
         sleep(0.5)
 
     socketio.emit("findProof", transaction)
 
+
 transactions = MiningPool(send_to_connected_clients, True)
 
+
 @app.route("/")
 def index():
     return "Hello From the Mining"
 
-@app.route("/queue", methods= ["POST"])
+
+@app.route("/queue", methods=["POST"])
 def add_data_to_queue():
     data = request.get_json()
-    
+
     if data is None:
         raise IncorrectPayloadException()
-    
+
     transactions.add_to_pool(data)
 
     return jsonify(success=True), HttpCode.OK.value
-    
 
-def valid_proof(hash_, nonce):
+
+def valid_proof(hash_, nonce) -> bool:
     valid = (hash_.startswith('0' * difficulty))
-    return valid and hash_ == sha256(
-        (str(nonce) + ongoing_transaction["to"] + ongoing_transaction["from"] + str(ongoing_transaction["amount"]) + ongoing_transaction["id"] + ongoing_transaction["signature"]).encode('utf-8')
-      ).hexdigest();
+    toHash = (str(nonce) + str(
+        ongoing_transaction["amount"]) + ongoing_transaction["id"] + ongoing_transaction["signature"])
+    toHash = toHash.replace("\n", "")
+    toHash = toHash.replace("\r", "")
+    computedHash = sha256(toHash.encode('utf-8')).hexdigest()
+    return valid and hash_ == computedHash
+
 
 @socketio.on('connect')
 def client_connect():
     global connected_clients
     connected_clients += 1
 
+
 @socketio.on('disconnect')
 def client_disconnect():
     global connected_clients
     connected_clients -= 1
 
+
 @socketio.on('echo')
 def handle(message):
     socketio.emit("response", message)
 
+
 @socketio.on('proof')
 def handle_proofs(message):
-    global ongoing_proof, ongoing_transaction, blockchain_url, blockchain_wallet_url, transactions
-    
+    global ongoing_proof, ongoing_transaction, blockchain_url, transactions
+    # print("ZACK OUTPUT - GETTING HERE 0", file=sys.stderr)
```

Comment:
Leaving these print statements here for now until the multiple findProof trigger bug is resolved